### PR TITLE
Feature/baip 468

### PR DESCRIPTION
**Pull request by anaim**

### CODE REVIEW

File: src/schema/failure-mode/failure-mode.factory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { CreateFailureModeInput } from './dto/create-failure-mode.input';
+import { FailureModeFactory } from './failure-mode.factory';
+import { domainFailureMode } from './__stubs__';
+import { FailureMode } from './models/failure-mode.model';
+
+describe('FailureModeFactory', () => {
+	test('CreateFailureMode() constructs an instance of an FailureMode GraphQL model', () => {
+		const result = FailureModeFactory.CreateFailureMode(domainFailureMode);
+		// const object = {
```
Comment (fortunato) [inline src/schema/failure-mode/failure-mode.factory.spec.ts:9]: Can this be removed?

File: src/schema/failure-mode/failure-mode.repository.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+			maintenanceRamsEnv,
+			maintenanceRamsP,
+			analysisRamsTotalPriority,
+			verificationRamsTotalPriority,
+			maintenanceRamsTotalPriority,
+			analysisRamsWeightedPriority,
+			verificationRamsWeightedPriority,
+			maintenanceRamsWeightedPriority,
+			failureModeType,
+			// copyOfFailureModeId,
```
Comment (fortunato) [inline src/schema/failure-mode/failure-mode.repository.ts:214]: What's up with these?

File: src/schema/failure-mode/failure-mode.resolver.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			expect(commandBusMock.execute).toHaveBeenCalledTimes(1);
+			expect(commandBusMock.execute).toHaveBeenCalledWith(new CreateFailureModeCommand(failureModeInput));
+
+			expect(result).toBeInstanceOf(FailureMode);
+			expect(typeof result.id).toBe('string');
+		});
+	});
+
+	describe('updateFailureMode', () => {
+		test('updates and returns an failureMode', async () => {
```
Comment (fortunato) [inline src/schema/failure-mode/failure-mode.resolver.spec.ts:49]: Typo, should be: "returns a failureMode"

File: src/schema/failure-mode/types/failure-mode.repository.interface.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		causeOfFailure: true,
+		sourceOfFailure: true,
+		consequenceOfFailure: true,
+		noticableFailure: true,
+	},
+});
+export type FailureMode = Prisma.failureModesGetPayload<typeof failureModes>;
+
+export interface IFailureModeRepository {
+	// findFailureModes(surveyId: string): Promise<FailureMode[]>;
```
Comment (fortunato) [inline src/schema/failure-mode/types/failure-mode.repository.interface.ts:63]: Can this be removed?

File: src/schema/ti/condition.factory.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import { Condition } from './models/condition.model';
+import { Condition as DomainCondition } from './types/condition.repository.interface';
+// import { CreateConditionInput } from './dto/create-condition.input';
```
Comment (fortunato) [inline src/schema/ti/condition.factory.ts:3]: Can this be removed?

File: src/schema/ti/models/defect.model.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	@Field((type) => String, { nullable: true })
+	ramsWeightedPriority?: string;
+
+	@Field((type) => String, { nullable: true })
+	createdAt?: string;
+
+	@Field((type) => String, { nullable: true })
+	updatedAt?: string;
+
+	// TODO Ensure the Condition model can be retrieved as a property
```
Comment (fortunato) [inline src/schema/ti/models/defect.model.ts:116]: I think this should be addressed. Can I ask you to make this happen?

**Review by fortunato — CHANGES_REQUESTED** [review]
I am not seeing any `@ResolveField` decorators yet for fetching the entities related to measures and cyclic measures. Will you be adding these now that we have a the models set up?

File: src/schema/failure-mode/models/failure-mode-meta-data.model.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { Field, ObjectType } from '@nestjs/graphql';
+
+@ObjectType({ description: 'failureModeMetaData' })
+export class FailureModeMetaData {
+	@Field((type) => String)
+	public faalOorzaak?: string;
+
+	@Field((type) => String)
+	public bronVanValen?: string;
```
Comment (bennyvdhoogen) [inline src/schema/failure-mode/models/failure-mode-meta-data.model.ts:9]: Typo here: 'bronVanValen' should be 'bronVanFalen' I suppose. See this in other places as well, but I suppose those are just refering to this one.

File: src/schema/failure-mode/models/failure-mode.model.spec.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+		failureMode.analysisRamsTotalPriority = null;
+		failureMode.verificationRamsTotalPriority = null;
+		failureMode.maintenanceRamsTotalPriority = null;
+		failureMode.analysisRamsWeightedPriority = null;
+		failureMode.verificationRamsWeightedPriority = null;
+		failureMode.maintenanceRamsWeightedPriority = null;
+		failureMode.copyOfFailureModeId = null;
+		failureMode.surveyScopeId = null;
+		failureMode.failureModeType = null;
+		// failureMode.function = null;
```
Comment (bennyvdhoogen) [inline src/schema/failure-mode/models/failure-mode.model.spec.ts:47]: Can be removed?

File: src/schema/failure-mode/models/failure-mode.model.spec.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+		failureMode.guideword = null;
+		// failureMode.failureMode = null;
```
Comment (bennyvdhoogen) [inline src/schema/failure-mode/models/failure-mode.model.spec.ts:49]: Can be removed?

File: src/schema/failure-mode/models/failure-mode.model.spec.ts [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+			analysisRamsTotalPriority: null,
+			verificationRamsTotalPriority: null,
+			maintenanceRamsTotalPriority: null,
+			analysisRamsWeightedPriority: null,
+			verificationRamsWeightedPriority: null,
+			maintenanceRamsWeightedPriority: null,
+			copyOfFailureModeId: null,
+			surveyScopeId: null,
+			failureModeType: null,
+			// function: null,
```
Comment (bennyvdhoogen) [inline src/schema/failure-mode/models/failure-mode.model.spec.ts:98]: Can be removed?

File: src/schema/failure-mode/models/failure-mode.model.spec.ts [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+			guideword: null,
+			// failureMode: null,
```
Comment (bennyvdhoogen) [inline src/schema/failure-mode/models/failure-mode.model.spec.ts:100]: Can be removed?

File: src/schema/survey/survey.repository.ts [hunk 12]
```diff
@@ -4,15 +4,14 @@ import { Injectable } from '@nestjs/common';
...
-import { Survey } from './models/survey.model';
+import { DbSurvey, ISurveyRepository } from './types/survey.repository.interface';
 import { CreateSurveyInput } from './dto/create-survey.input';
 
 @Injectable()
 export class SurveyRepository implements ISurveyRepository {
 	public constructor(private readonly prisma: PrismaService) {}
 
-	async createSurvey(input: CreateSurveyInput): Promise<Survey> {
+	async createSurvey(input: CreateSurveyInput): Promise<DbSurvey> {
```
Comment (bennyvdhoogen) [inline src/schema/survey/survey.repository.ts:14]: Why this name change tot DbSurvey? I understand the distinction you want to make between the Model Survey and the raw database output from the `surveys` table within this repository, but shouldn't the repository make sure that the output is the model Survey?

File: src/schema/ti/condition.resolver.spec.ts [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+	// 		createdAt: domainCondition.created_at ?? null,
+	// 		updatedAt: domainCondition.updated_at ?? null,
+	// 		craHistoryUnityCheck: 10.2,
+	// 		craInitialUnityCheck: 10.2,
+	// 	};
+	// 	delete object.created_at;
+	// 	delete object.updated_at;
+	// 	expect(conditions).toBeInstanceOf(Condition);
+	// 	expect(conditions).toEqual(object);
+	// });
```
Comment (bennyvdhoogen) [inline src/schema/ti/condition.resolver.spec.ts:76]: Can you remove this if it is unused?

**Review by bennyvdhoogen — COMMENTED** [review]
Looks good! Few minor remarks.

**Review by bennyvdhoogen — APPROVED** [review]

**Pull request merged**